airflow/api_connexion/endpoints/variable_endpoint.py

```diff
@@ -14,41 +14,83 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import List, Optional
 
-# TODO(mik-laj): We have to implement it.
-#     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8142
+from flask import Response, request
+from marshmallow import ValidationError
 
+from airflow.api_connexion.exceptions import BadRequest, NotFound
+from airflow.api_connexion.schemas.variable_schema import variable_collection_schema, variable_schema
+from airflow.models import Variable
+from airflow.utils.session import provide_session
 
-def delete_variable():
+
+def delete_variable(variable_key: str) -> Response:
     """
     Delete variable
     """
-    raise NotImplementedError("Not implemented yet.")
+    if Variable.delete(variable_key) == 0:
+        raise NotFound("Variable not found")
+    return Response(status=204)
 
 
-def get_variable():
+def get_variable(variable_key: str) -> Response:
     """
     Get a variables by key
     """
-    raise NotImplementedError("Not implemented yet.")
+    try:
+        var = Variable.get(variable_key)
+    except KeyError:
+        raise NotFound("Variable not found")
+    return variable_schema.dump({"key": variable_key, "val": var})
 
 
-def get_variables():
+@provide_session
+def get_variables(session, limit: Optional[int], offset: Optional[int] = None) -> Response:
     """
     Get all variable values
     """
-    raise NotImplementedError("Not implemented yet.")
+    query = session.query(Variable)
```

Comment:
```suggestion
    query = session.query(Variable).order_by(Variable.id)
```
This will allow continuity of items between pages.